### PR TITLE
SHA256 add FixedString support

### PR DESCRIPTION
**Pull request by ucasfl**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Add FixedString support in Hashing functions.

Detailed description / Documentation draft:

...

By adding documentation, you'll allow users to try your new feature immediately, not when someone else will have time to document it later. Documentation is necessary for all features that affect user experience in any way. You can add brief documentation draft above, or add documentation right into your patch as Markdown files in [docs](https://github.com/ClickHouse/ClickHouse/tree/master/docs) folder.

If you are doing this for the first time, it's recommended to read the lightweight [Contributing to ClickHouse Documentation](https://github.com/ClickHouse/ClickHouse/tree/master/docs/README.md) guide first.

### CODE REVIEW

**Comment by qoega** [comment]
Can you add relevant test cases?

File: tests/queries/0_stateless/01338_sha256_fixedstring.sql [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+INSERT INTO defaults SELECT randomFixedString(20);
+INSERT INTO defaults SELECT randomFixedString(20);
+INSERT INTO defaults SELECT randomFixedString(20);
+INSERT INTO defaults SELECT randomFixedString(20);
+INSERT INTO defaults SELECT randomFixedString(20);
+INSERT INTO defaults SELECT randomFixedString(20);
+INSERT INTO defaults SELECT randomFixedString(20);
+INSERT INTO defaults SELECT randomFixedString(20);
+INSERT INTO defaults SELECT randomFixedString(20);
+INSERT INTO defaults SELECT randomFixedString(20);
```
Comment (qoega) [inline tests/queries/0_stateless/01338_sha256_fixedstring.sql:19]:
```suggestion
INSERT INTO defaults SELECT s
FROM generateRandom('s FixedString(20)', 1, 1, 1)
LIMIT 20
```

File: tests/queries/0_stateless/01338_sha256_fixedstring.sql [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+SELECT SHA256(s) FROM defaults;
```
Comment (qoega) [inline tests/queries/0_stateless/01338_sha256_fixedstring.sql:21]:
```suggestion
SELECT hex(SHA256(s)) FROM defaults;
```

File: tests/queries/0_stateless/01338_sha256_fixedstring.sql (hunk 1 above)
Comment (qoega) [reply tests/queries/0_stateless/01338_sha256_fixedstring.sql:19]: randomFixedString is not stable between different calls. You can use predefined seed in `generateRandom` instead.

Comment (ucasFL) [reply tests/queries/0_stateless/01338_sha256_fixedstring.sql]: ok, i will change it.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**